relayer-engine/src/executor/utils.ts

```diff
@@ -0,0 +1,462 @@
+      fields: [
+        ["key", "u8"],
+        ["updateAuthority", "pubkeyAsString"],
+        ["mint", "pubkeyAsString"],
+        ["data", Data],
+        ["primarySaleHappened", "u8"], // bool
+        ["isMutable", "u8"], // bool
+      ],
+    },
+  ],
+  [
+    EditionMarker,
+    {
+      kind: "struct",
+      fields: [
+        ["key", "u8"],
+        ["ledger", [31]],
+      ],
+    },
+  ],
+]);
+
+export const extendBorsh = () => {
+  (BinaryReader.prototype as any).readPubkey = function () {
+    const reader = this as unknown as BinaryReader;
+    const array = reader.readFixedArray(32);
+    return new PublicKey(array);
+  };
+
+  (BinaryWriter.prototype as any).writePubkey = function (value: PublicKey) {
+    const writer = this as unknown as BinaryWriter;
+    writer.writeFixedArray(value.toBuffer());
+  };
+
+  (BinaryReader.prototype as any).readPubkeyAsString = function () {
+    const reader = this as unknown as BinaryReader;
+    const array = reader.readFixedArray(32);
+    return base58.encode(array) as StringPublicKey;
+  };
+
+  (BinaryWriter.prototype as any).writePubkeyAsString = function (
+    value: StringPublicKey
+  ) {
+    const writer = this as unknown as BinaryWriter;
+    writer.writeFixedArray(base58.decode(value));
+  };
+};
+
+extendBorsh();
+
+export const decodeMetadata = (buffer: Buffer): Metadata => {
+  const metadata = deserializeUnchecked(
+    METADATA_SCHEMA,
+    Metadata,
+    buffer
+  ) as Metadata;
+  metadata.data.name = metadata.data.name.replace(METADATA_REPLACE, "");
+  metadata.data.uri = metadata.data.uri.replace(METADATA_REPLACE, "");
+  metadata.data.symbol = metadata.data.symbol.replace(METADATA_REPLACE, "");
+  return metadata;
+};
+
+export const getMetadataAddress = async (
+  mintKey: string
+): Promise<[PublicKey, number]> => {
+  const seeds = [
+    Buffer.from("metadata"),
+    new PublicKey(METADATA_PROGRAM_ID).toBuffer(),
+    new PublicKey(mintKey).toBuffer(),
+  ];
+  return PublicKey.findProgramAddress(
+    seeds,
+    new PublicKey(METADATA_PROGRAM_ID)
+  );
+};
+
+export const getMetaplexData = async (
+  mintAddresses: string[],
+  chainInfo: ChainConfigInfo
+) => {
+  const promises = [];
+  for (const address of mintAddresses) {
+    promises.push(getMetadataAddress(address));
+  }
+  const metaAddresses = await Promise.all(promises);
+  const connection = new Connection(chainInfo.nodeUrl, "confirmed");
+  const results = await getMultipleAccountsRPC(
+    connection,
+    metaAddresses.map((pair) => pair && pair[0])
+  );
+
+  const output = results.map((account) => {
+    if (account === null) {
+      return undefined;
+    } else {
+      if (account.data) {
+        try {
+          const MetadataParsed = decodeMetadata(account.data);
+          return MetadataParsed;
+        } catch (e) {
+          // console.error(e);
+          return undefined;
+        }
+      } else {
+        return undefined;
+      }
+    }
+  });
+
+  return output;
+};
```

Comment:
This entire file should be deleted. We don't care about Metaplex NFT data and even if we did we ought to use Metaplex's SDK.

Reply:
After digging around a bit, I found out that we are only using Metaplex sdk to extract the token symbol and nothing more (which is marked as a required field in a `WalletBalance`).
Probably according also to [this](https://solana.stackexchange.com/questions/1900/get-token-metadata-decimals-symbol-name).

Just thought I'd update you on the status of this task before ripping out the code for Metaplex. What do you think?